server/src/main/java/io/druid/curator/inventory/CuratorInventoryManager.java

```diff
@@ -257,9 +276,19 @@ public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) th
         case CHILD_UPDATED:
           synchronized (lock) {
             final ChildData child = event.getData();
+
+            byte[] data = getZkDataForNode(child.getPath());
```

Comment:
"curatorFramework.getData().decompressed().forPath(path)" should always get latest data from ZK.